src/main/java/core/basesyntax/service/impl/DeepCopyImpl.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.FruitTransaction;
+import core.basesyntax.service.DeepCopy;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DeepCopyImpl implements DeepCopy {
+    @Override
+    public List<FruitTransaction> getDeepCopy(List<FruitTransaction> list) {
+        List<FruitTransaction> copyList = new ArrayList<>(list.size());
+        for (FruitTransaction fruit : list) {
+            copyList.add(new FruitTransaction(fruit.getType(),
+                    fruit.getFruit(), fruit.getQuantity()));
+        }
+        return copyList;
+    }
```

Comment:
Ensure that the copy constructor of FruitTransaction actually creates a deep copy of the object. If any of the fields within FruitTransaction are mutable and not primitive or immutable types, a new instance of those fields should be created to prevent shared references with the original object's fields.